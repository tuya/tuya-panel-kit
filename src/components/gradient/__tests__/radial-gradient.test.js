import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import RadialGradient from '../radial-gradient';

const radialGradient = {
  stops: [{
    offset: '0%',
    stopColor: 'red',
    stopOpacity: '1',
  }, {
    offset: '20%',
    stopColor: 'yellow',
    stopOpacity: '1',
  }, {
    offset: '40%',
    stopColor: 'blue',
    stopOpacity: '1',
  }],
  rx: '70%',
  ry: '90%',
  cx: '50%',
};

describe('RadialGradient Component', () => {
  it('basic render', () => {
    const component = renderer.create(
      <RadialGradient
        {...radialGradient}
      />
    );

    expect(component.root.instance.props.stops).toEqual(radialGradient.stops);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
