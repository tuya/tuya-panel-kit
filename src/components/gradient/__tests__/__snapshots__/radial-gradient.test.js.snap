// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadialGradient Component basic render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "flex": 1,
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      },
      null,
    ]
  }
>
  <RNSVGSvgView
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        null,
        false,
        Object {
          "flex": 0,
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <RNSVGDefs>
      <RNSVGRadialGradient
        cx="50%"
        cy="50%"
        fx="50%"
        fy="50%"
        gradient={
          Array [
            1,
            0,
            0,
            1,
            1,
            1,
            0,
            1,
            0,
            0,
            1,
            1,
            0,
            0.2,
            0.4,
          ]
        }
        gradientTransform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        gradientUnits={0}
        name="radial-gradient"
        rx="70%"
        ry="90%"
      />
    </RNSVGDefs>
    <RNSVGRect
      fill={
        Array [
          1,
          "radial-gradient",
        ]
      }
      fillOpacity={1}
      fillRule={1}
      height="1334"
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      propList={
        Array [
          "fill",
        ]
      }
      rx="0"
      ry="0"
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
      width="750"
      x="0"
      y="0"
    />
  </RNSVGSvgView>
</View>
`;
